modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/plugins/fields-sidebar/components/FieldsSidebar.js

```diff
@@ -0,0 +1,205 @@
+	useFormState,
+} from 'dynamic-data-mapping-form-renderer';
+import React, {useState} from 'react';
+
+import {
+	EDIT_CUSTOM_OBJECT_FIELD,
+	UPDATE_DATA_DEFINITION_AVAILABLE_LANGUAGE,
+	dropLayoutBuilderField,
+} from '../../../../js/actions.es';
+import Sidebar from '../../../../js/components/sidebar/Sidebar.es';
+import {EVENT_TYPES} from '../../../eventTypes';
+import FieldsSidebarBody from './FieldsSidebarBody';
+import FieldsSidebarSettingsBody from './FieldsSidebarSettingsBody';
+import FieldsSidebarSettingsHeader from './FieldsSidebarSettingsHeader';
+
+const sortFieldTypes = (fieldTypes) =>
+	fieldTypes.sort(({displayOrder: a}, {displayOrder: b}) => a - b);
+
+export const DataEngineFieldsSidebar = ({title}) => {
+	const {
+		activePage,
+		config,
+		customFields,
+		dataDefinition,
+		dataLayout,
+		defaultLanguageId,
+		editingLanguageId,
+		focusedCustomObjectField,
+		focusedField,
+		pages,
+	} = useFormState();
+	const {fieldTypes} = useConfig();
+	const dispatch = useForm();
+
+	const hasFocusedCustomObjectField = (focusedCustomObjectField) => {
+		return !!focusedCustomObjectField.settingsContext;
+	};
+
+	const hasFocusedField = Object.keys(focusedField).length > 0;
+
+	const displaySettings =
+		hasFocusedCustomObjectField(focusedCustomObjectField) ||
+		hasFocusedField;
+
+	const sortedFieldTypes = sortFieldTypes(
+		fieldTypes.filter(({group}) => group === 'basic')
+	);
+
+	return (
+		<FieldsSidebar
+			config={config}
+			customFields={customFields}
+			dataLayout={dataLayout}
+			defaultLanguageId={defaultLanguageId}
+			dispatchEvent={(type, payload) => {
+				if (type === 'field_change') {
+					const {editingLanguageId} = payload;
+
+					if (
+						!dataDefinition.availableLanguageIds.includes(
+							editingLanguageId
+						)
+					) {
+						dispatch({
+							payload: editingLanguageId,
+							type: UPDATE_DATA_DEFINITION_AVAILABLE_LANGUAGE,
+						});
+					}
+
+					if (hasFocusedCustomObjectField(focusedCustomObjectField)) {
+						dispatch({payload, type: EDIT_CUSTOM_OBJECT_FIELD});
+
+						return;
+					}
+				}
+
+				if (!hasFocusedCustomObjectField(focusedCustomObjectField)) {
+					dispatch({
+						payload,
+						type,
+					});
+				}
+			}}
+			displaySettings={displaySettings}
+			editingLanguageId={editingLanguageId}
+			fieldTypes={sortedFieldTypes}
+			focusedCustomObjectField={focusedCustomObjectField}
+			focusedField={focusedField}
+			hasFocusedCustomObjectField={hasFocusedCustomObjectField}
+			onClick={() => {
+				dispatch({
+					type: EVENT_TYPES.SIDEBAR.FIELD.BLUR,
+				});
+			}}
+			onDoubleClick={({name}) => {
+				dispatch({
+					payload: dropLayoutBuilderField({
+						addedToPlaceholder: true,
+						fieldTypeName: name,
+						fieldTypes,
+						indexes: {
+							columnIndex: 0,
+							pageIndex: activePage,
+							rowIndex: pages[activePage].rows.length,
+						},
+					}),
+					type: CORE_EVENT_TYPES.FIELD.ADD,
+				});
+			}}
+			title={title}
+		/>
+	);
+};
```

Comment:
As the reason for decoupling this was before to allow `FieldsSidebar` to be used within Forms with its own data, I think we don’t need it anymore, we can move into those components now.